components/TodoModal.vue

```diff
@@ -0,0 +1,109 @@
+                            </div>
+  
+                            <div class="w-full mb-2">
+                                <label class="block text-gray-700 text-sm font-bold mb-2" for="date">
+                                    Date
+                                </label>
+                                <div class="dp-container-wrap shadow appearance-none border rounded">
+                                    <VueDatePicker 
+                                        v-model="date"
+                                        teleport
+                                        auto-apply
+                                        input-class-name="dp-custom-input"
+                                        :placeholder="data?.date?.seconds?new Date(data?.date?.seconds*1000):'Select date'"
+                                    />
+                                </div>
+                            </div>
+                            <div class="mb-4 pt-6">            
+                                <label class="inline-flex items-center pl-1 cursor-pointer">
+                                <input type="checkbox" value="" class="sr-only peer" :checked="data?.done??false" v-model="done">
+                                <div class="relative w-11 h-6 bg-gray-400 rounded-full peer dark:bg-gray-700 peer-focus:ring-4 peer-focus:ring-orange-300 dark:peer-focus:ring-orange-800 peer-checked:after:translate-x-full rtl:peer-checked:after:-translate-x-full peer-checked:after:border-white after:content-[''] after:absolute after:top-0.5 after:start-[2px] after:bg-white after:border-gray-300 after:border after:rounded-full after:h-5 after:w-5 after:transition-all dark:border-gray-600 peer-checked:bg-orange-500"></div>
+                                <span class="ms-3 text-sm font-medium text-gray-900 dark:text-gray-300">Done</span>
+                                </label>
+                            </div>
+
+                            <div class="flex items-center flex-row-reverse">
+                                <button class="bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded focus:outline-none focus:shadow-outline" type="button" @click="saveData">
+                                    Save
+                                </button>
+                            </div>
+                        </div>
+                        <button class="text-xs mt-2 bg-transparent hover:bg-red-500 text-red-700 hover:text-white border border-red-500 hover:border-transparent rounded-full h-8 w-8 pb-1 flex items-center justify-center" @click="$emit('close')">
+                            &#9587;
+                        </button>
+                </div>
+            </div>
+            </div>
+        </teleport>    
+</template>
+
+<script setup lang="ts">
+import VueDatePicker from '@vuepic/vue-datepicker';
+import '@vuepic/vue-datepicker/dist/main.css'
+const emit = defineEmits<{
+  (event: 'saveData', payload: any): void;
```

Comment:
You can use new TS-friendly syntax for defineEmits https://vuejs.org/guide/typescript/composition-api#typing-component-emits, see the last example for 3.3+ version